ambry-server/src/test/java/com/github/ambry/server/AmbryServerRequestsTest.java

```diff
@@ -939,6 +941,60 @@ public void crossColoMetricsUpdateTest() throws Exception {
     responseList.forEach(Response::release);
   }
 
+  /**
+   * Tests for appropriate responses from healthCheck admin endpoint
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void healthCheckTest() throws InterruptedException, IOException {
+
+    //retrieves writeable partitions to test with
+    List<? extends PartitionId> partitionIds = clusterMap.getWritablePartitionIds(DEFAULT_PARTITION_CLASS);
+
+    for (PartitionId id : partitionIds) {
+
+      int correlationId = TestUtils.RANDOM.nextInt();
+      String clientId = TestUtils.getRandomString(10);
+
+      // Test 1: "Good" response excepted where the host state is in either Standby or Leader
+      HashMap<ReplicaId,ReplicaState> replicaToOriginalState = new HashMap<>(); //remembers original state of replica
+
+      //change all replica states in partition to Standby or Leader
+      for(ReplicaId replica : id.getReplicaIds()) {
+        replicaToOriginalState.put(replica, storageManager.getStore(replica.getPartitionId()).getCurrentState());
+        storageManager.getStore(replica.getPartitionId()).setCurrentState(
+            TestUtils.RANDOM.nextInt() % 2 == 0 ? ReplicaState.STANDBY : ReplicaState.LEADER);
+      }
+
+      AdminRequest adminRequest = new AdminRequest(AdminRequestOrResponseType.HealthCheck, id, correlationId, clientId);
+      AdminResponseWithContent response =
+          (AdminResponseWithContent) sendRequestGetResponse(adminRequest, ServerErrorCode.No_Error);
+
+      assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
+      assertEquals("Payload was expected to be {\"health\":\"GOOD\"}","{\"health\":\"GOOD\"}",
+          new String(response.getContent(),StandardCharsets.UTF_8));
+
+      //Test 2: "BAD" response expected where change replicas to error State on this partition
+      //change all replica states in partition to ERROR
+      for(ReplicaId replica : id.getReplicaIds()) {
+        storageManager.getStore(replica.getPartitionId()).setCurrentState(ReplicaState.ERROR);
+      }
+
```

Comment:
replica.getPartitionId() would always return the same PartitionId in this for loop since the all the replicas belong to the same partition. We could just do
```
storageManager.getStore(id).setCurrentState(ReplicaState.ERROR);
```

Reply:
Yeah, thanks for that. I didn't realize at the time.